### DISCUSSION TITLE
How do I fetch Multiple Specific rows in a table from an Array?

### DISCUSSION THREAD

**Discussion by ethanfox** [supabase/supabase#12972] [Questions] [answered]
I have 2 tables Clients and Contacts

I have a table with a cell with specific user IDs in the client table called "contacts"   
![image](https://user-images.githubusercontent.com/22962600/224578787-c955f3ce-fb2b-4d1f-9b7c-d03a1cb59950.png)

How do I fetch the specific rows from the contacts table from the contacts array found in the client table?

Getting the array is easy enough

```
let associatedContactsArray = [];

async function getAssociatedContactsID() {
		const { data: contacts, error } = await supabaseClient
			.from('clients')
			.select('contacts')
			.eq('id', $clientID);
		if (contacts) {
			const [fetchedContacts] = contacts;
			if (fetchedContacts.contacts != null) {
				associatedContactsArray = fetchedContacts.contacts;
				hasAssociatedContacts = true;
				console.log('ASSOCIATED CONTACTS', associatedContactsArray);

				getAssociatedContacts();
			} else {
				console.log('THERE ARE NO CONTACTS');
				hasAssociatedContacts = false;
			}
		}

		if (error) {
			console.log('ERROR GETTING ASSOCIATED CONTACTS', error);
		}
	}
```

The following does not work.

    `async function getAssociatedContacts() {
		const { data: contacts, error } = await supabaseClient
			.from('contacts')
			.select('*')
			.filter('id', 'in', { associatedContactsArray });
		if (contacts) {
			const [fetchedContacts] = contacts;
			if (fetchedContacts.contacts != null) {
				console.log('GOT CONTACT DETAILS');
				console.log('CONTACTS', contacts);
			} else {
				console.log('NO CONTACT DETAILS');
				hasAssociatedContacts = false;
			}
		}

		if (error) {
			console.log('ERROR GETTING ASSOCIATED CONTACTS', error);
		}
	}`

I get the error:
`    {
    "code": "PGRST100",
    "details": "unexpected \"[\" expecting \"(\"",
    "hint": null,
    "message": "\"failed to parse filter (in.[object Object])\" (line 1, column 4)"
}`

Referencing Match the Filter: https://supabase.com/docs/reference/javascript/filter

**Comment by GaryAustin1** [comment]
Just do `.in('id',associatedContactsArray)`.

Otherwise per the docs it is `.filter('id', 'in', '("id1","id2","id3")')`
in works on a list in postgres not an array.  `{  }` is postgres array string.
https://supabase.com/docs/reference/javascript/filter

**Reply by ethanfox** [reply]
It now with ends with no response - `NO CONTACT DETAILS` The array is an array of strings, and the id column is the `uuid` type. 

This is the console.log of the array.
<img width="345" alt="Screenshot 2023-03-14 at 6 16 15 PM" src="https://user-images.githubusercontent.com/22962600/225163022-ed69e265-ac8d-485b-a8f5-dfd3084b887d.png">

**Reply by GaryAustin1** [reply]
You should show your call and if possible your url begin generated.
Do you have RLS enabled for and meet the select policy on that table?

**Reply by ethanfox** [reply]
RLS is enabled. I have "Enable read access for all users" on the table (The one provided by Supabase) I'm not sure how get the URL generated by the call.

**Reply by ethanfox** [reply]
I think I'll just create a separate table to solve this problem. 

Cell 1 : ClientID Cell2 2: ContactID -> then just query the table and select rows equal to ClientID

**Comment by ethanfox** [comment, marked as answer]
### Solution

if (data) will not get tripped in this scenario. Which is the typical format for fetching data from Supabase as of early 2023.

With `in` there's clearly an assumption of data. **IF** the data returns as an empty array - it will best be handled in another function

Correct Code:

```
let associatedContactsArray = [String];

// 'id' is of the uuid column type

async function getAssociatedContacts() {
		const { data, error } = await supabaseClient
			.from('contacts')
			.select()
			.in('id', associatedContactsArray);
		if (error) {
			console.log('ERROR GETTING ASSOCIATED CONTACTS', error);
		} else {
			console.log('GOT CONTACT DETAILS');
			console.log('CONTACT DETAILS FETCHED', data);
		}
	}
```
